tensorflow/python/ops/linalg_ops.py

```diff
@@ -305,6 +305,49 @@ def _composite_impl(matrix, rhs, l2_regularizer):
     return gen_linalg_ops.matrix_solve_ls(
         matrix, rhs, l2_regularizer, fast=fast, name=name)
 
+@tf_export('linalg.eig', v1=[])
+def eig(tensor, name=None):
+  """Computes the eigen decomposition of a batch of matrices.
+
+  Computes the eigenvalues and eigenvectors of the innermost N-by-N matrices
+  in `tensor` such that
+  `tensor[...,:,:] * v[..., :,i] = e[..., i] * v[...,:,i]`, for i=0...N-1.
+
```

Comment:
We need to mention that the eigenvalues and eigenvectors for a non-Hermitian matrix in general are complex, and the eigenvectors are not guaranteed to be linearly independent (The matrix may be "defective", i.e. have lower geometric than algebraic multiplicity). Furthermore, mention that what we compute here are the _right_ eigenvectors. Maybe add a reference to R. A. Horn & C. R. Johnson, "Matrix Analysis", Chapter 1.